code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -331,6 +341,13 @@
 			recording_recipe = null
 			. = TRUE
 
+		if("reaction_lookup")
+			//drink dispensers shouldnt be able to look up reagents
+			if(istype(src, /obj/machinery/chem_dispenser/drinks))
```

Comment:
Janky dm version of https://github.com/BeeStation/BeeStation-Hornet/pull/8579/files#r1559864976 .

That option is preferable.

Reply:
If you want to check for a specific type, but not its subtypes you can use `type == /type/path` instead of `istype(type, /type/path)`

Reply:
> If you want to check for a specific type, but not its subtypes you can use `type == /type/path` instead of `istype(type, /type/path)`.

I do want drinks and their subtypes though, specifically.


> Ideally UI elements are unaware of the DM types that are using them, since they are meant to be independant from the game apart from the data transferred.

Thats disappointing. I rather would have the UI element to lookup just not appear at all for drink dispensers if it cannot be used, but if thats the way it has to be, alright.